rocketmq-spring-boot/src/main/java/org/apache/rocketmq/spring/core/RocketMQTemplate.java

```diff
@@ -587,11 +590,19 @@ public boolean createAndStartTransactionMQProducer(String txProducerGroup, Rocke
         return true;
     }
 
-    private TransactionMQProducer createTransactionMQProducer(String name, RocketMQLocalTransactionListener transactionListener,
-                                                              ExecutorService executorService) {
+    private TransactionMQProducer createTransactionMQProducer(String name,
+                                                              RocketMQLocalTransactionListener transactionListener,
+                                                              ExecutorService executorService, RPCHook rpcHook) {
         Assert.notNull(producer, "Property 'producer' is required");
         Assert.notNull(transactionListener, "Parameter 'transactionListener' is required");
-        TransactionMQProducer txProducer = new TransactionMQProducer(name);
+        TransactionMQProducer txProducer;
+        if (Objects.nonNull(rpcHook)) {
+            txProducer = new TransactionMQProducer(name, rpcHook);
+            txProducer.setVipChannelEnabled(false);
+            txProducer.setInstanceName(RocketMQUtil.getInstanceName(rpcHook, name));
```

Comment:
Different instanceName is using for different ak/sk in multi listeners annotated by @RocketMQTransactionListener.